x-pack/plugins/upgrade_assistant/server/lib/es_deprecations_status.test.ts

```diff
@@ -6,14 +6,96 @@
 import fakeDeprecations from './__fixtures__/fake_deprecations.json';
+import * as healthIndicatorsMock from './__fixtures__/health_indicators';
+
 import type { FeatureSet } from '../../common/types';
 const fakeIndexNames = Object.keys(fakeDeprecations.index_settings);
 
+describe('getHealthIndicators', () => {
+  let esClient: ScopedClusterClientMock;
+  beforeEach(() => {
+    esClient = elasticsearchServiceMock.createScopedClusterClient();
+  });
+
+  it('returns empty array on green indicators', async () => {
+    esClient.asCurrentUser.healthReport.mockResponse({
+      cluster_name: 'mock',
+      indicators: {
+        disk: healthIndicatorsMock.diskIndicatorGreen,
+        // @ts-ignore
+        shards_capacity: healthIndicatorsMock.shardCapacityIndicatorGreen,
+      },
+    });
+
+    const result = await getHealthIndicators(esClient);
+    expect(result).toEqual([]);
+  });
+
+  it('returns unhealthy shards_capacity indicator', async () => {
+    esClient.asCurrentUser.healthReport.mockResponse({
+      cluster_name: 'mock',
+      indicators: {
+        disk: healthIndicatorsMock.diskIndicatorGreen,
+        // @ts-ignore
+        shards_capacity: healthIndicatorsMock.shardCapacityIndicatorRed,
+      },
+    });
+
+    const result = await getHealthIndicators(esClient);
+    expect(result).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "correctiveAction": Object {
+            "action": "Increase the value of [cluster.max_shards_per_node] cluster setting or remove data indices to clear up resources.",
+            "cause": "Elasticsearch is about to reach the maximum number of shards it can host, based on your current settings.",
+            "impacts": Array [
+              Object {
+                "description": "The cluster has too many used shards to be able to upgrade.",
+                "id": "elasticsearch:health:shards_capacity:impact:upgrade_blocked",
+                "impact_areas": "[Array]",
+                "severity": 1,
+              },
+              Object {
+                "description": "The cluster is running low on room to add new shards. Adding data to new indices is at risk",
+                "id": "elasticsearch:health:shards_capacity:impact:creation_of_new_indices_blocked",
+                "impact_areas": "[Array]",
+                "severity": 1,
+              },
+            ],
+            "type": "healthIndicator",
+          },
+          "details": "Cluster is close to reaching the configured maximum number of shards for data nodes.",
+          "isCritical": true,
+          "message": "Elasticsearch is about to reach the maximum number of shards it can host, based on your current settings.",
+          "resolveDuringUpgrade": false,
+          "type": "health_indicator",
+          "url": "https://ela.st/fix-shards-capacity",
+        },
+      ]
+    `);
+  });
+
+  // TODO: Check with ES team the red status payload.
```

Comment:
Is this something you plan to follow up on post-merge?

Reply:
@Bamieh i already did check with @HiDAl and it looks correct :+1: I'll remove the comment